ansible_base/rbac/permission_registry.py

```diff
@@ -46,7 +47,7 @@ def register(self, *args, parent_field_name='organization'):
     def track_relationship(self, cls, relationship, role_name):
         self._tracked_relationships.add((cls, relationship, role_name))
 
-    def get_parent_model(self, model) -> Optional[Model]:
+    def get_parent_model(self, model) -> Union[type, None]:
```

Comment:
This is the same as `Optional` which we've used in other places. Should we try to standardize on either `Optional(type)` or `Union[type, None]` in the code?

Reply:
What you're proposing is `Optional[type]` and there be dragons there, because of the difference between `NoneType` and `None`. Looking for a solution, the first thing I stumbled on was `"Model"` as a string, which would specify the type, not an object of the type, also `typing.Type` was an option. There's also a _lot_ of differences in recent python versions, and although AWX is on 3.11, I don't want to assume this version if I don't have to. I'd rather make it less picky that force a certain python version. My issue with a lot of these approaches it that union/optional typing seemed to want both options to be a class, resulting in an expectation of `NoneType`.

If we were going to standardize on something, I would probably go for `Union[type, None]`, because this is brutal and reliable. This also isn't tremendously common. You will tend to hit this with _registries_ which do work crawling a collection of _classes_.

Reply:
I'm really struggling to grasp this. `Union[type, None]` and `Optional[type]` should absolutely be the same thing, and I have a strong preference to `Optional`.

Even the python `typing` [docs](https://docs.python.org/3/library/typing.html#typing.Optional) say:

>  `typing.Optional`
>
>    `Optional[X]` is equivalent to `X | None` (or `Union[X, None]`).

Is typeguard breaking spec here?

Reply:
I'll put that in, but I'm starting to think it was a version-specific behavior now.

Reply:
alright, I guess it's working...